plugins/modules/ec2_vpc_vgw.py

```diff
@@ -198,39 +212,50 @@ def get_vgw_info(vgws):
         return vgw_info
 
 
-def wait_for_status(client, module, vpn_gateway_id, status):
+def wait_for_status(
+    client, module: AnsibleAWSModule, vpn_gateway_id: Union[str, List[str]], desired_status: str
+) -> Tuple[bool, Any]:
     polling_increment_secs = 15
     max_retries = module.params.get("wait_timeout") // polling_increment_secs
-    status_achieved = False
+    try:
+        wait_for_resource_state(client, module, "vpn_gateway_exists", VpnGatewayIds=vpn_gateway_id)
+        if desired_status == "attached":
+            wait_for_resource_state(
+                client,
+                module,
+                "vpn_gateway_attached",
+                VpnGatewayIds=vpn_gateway_id,
+                delay=polling_increment_secs,
+                max_attempts=max_retries,
+            )
+        elif desired_status == "detached":
+            wait_for_resource_state(
+                client,
+                module,
+                "vpn_gateway_detached",
+                VpnGatewayIds=vpn_gateway_id,
+                delay=polling_increment_secs,
+                max_attempts=max_retries,
+            )
```

Comment:
```suggestion
        if desired_status not in ("attached", "detached"):
              module.fail_json(msg=f"Unsupported status: {desired_status}")
        wait_for_resource_state(
                client,
                module,
                f"vpn_gateway_{desired_status}",
                VpnGatewayIds=vpn_gateway_id,
                delay=polling_increment_secs,
                max_attempts=max_retries,
            )
       
```

Reply:
changing VpnGatewayIds=vpn_gateway_id to VpnGatewayIds=[vpn_gateway_id] causes failure with
```
"msg": "An exception happened while trying to wait for 'vpn_gateway_attached': Parameter validation failed:\nInvalid type for parameter VpnGatewayIds[0], value: ['vgw-0c3df60baf0acc49f'], type: <class 'list'>, valid types: <class 'str'>",
    "resource_actions": [
        "ec2:DescribeVpnGateways",
        "ec2:CreateVpnGateway",
        "ec2:AttachVpnGateway"
    ]
```
even though API states otherwise it to be a list of str, weird.

Reply:
Same as below.

Reply:
Hi @alinabuzachis , fixed. It was indeed being passed as List[Str] at the API call.
Only thing was that it was handled to be passed as List[Str] earlier in the function call tree.
Updated the code to ensure it is passed as List[Str] before calling module_utils/ec2 functions.
https://github.com/ansible-collections/community.aws/pull/2171/commits/27b07e056bccfd390b8c1425acfdaa6ac58a44af